_posts/2021-02-26-feather-nrf52840.md

```diff
@@ -0,0 +1,462 @@
+ *
+ * @file
+ * @brief       Configuration of SAUL mapped GPIO pins
+ *
+ * @author      Martine S. Lenders <m.lenders@fu-berlin.de>
+ */
+
+#ifndef GPIO_PARAMS_H
+#define GPIO_PARAMS_H
+
+#include "board.h"
+#include "saul/periph.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief    LED configuration
+ */
+static const  saul_gpio_params_t saul_gpio_params[] =
+{
+    {
+        .name  = "LED Red (D3)",
+        .pin   = LED0_PIN,
+        .mode  = GPIO_OUT,
+        .flags = (SAUL_GPIO_INIT_CLEAR),
+    },
+    {
+        .name  = "LED Blue (Conn)",
+        .pin   = LED1_PIN,
+        .mode  = GPIO_OUT,
+        .flags = (SAUL_GPIO_INIT_CLEAR),
+    },
+    {
+        .name  = "UserSw",
+        .pin   = BTN0_PIN,
+        .mode  = BTN0_MODE,
+        .flags = SAUL_GPIO_INVERTED,
+    },
+};
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* GPIO_PARAMS_H */
+/** @} */
+{% endhighlight %}
+
+Lastly, the board needs to define some low-level debugging LED and button [GPIO]
+pins and needs to initialize those pins. We do this in the in
+[`boards/feather-nrf52840/include/board.h`][board.h] and
+[`boards/feather-nrf52840/board.c`][board.c].
+Rumors tell, this might not be necessary in the future, so watch out for this
+later if you read this when you follow my instructions ;-).
+
+{% highlight C %}
+/*
+ * Copyright (C) 2020 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include "cpu.h"
+#include "board.h"
+
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the boards LEDs */
+    gpio_init(LED0_PIN, GPIO_OUT);
+    gpio_clear(LED0_PIN);
+    gpio_init(LED1_PIN, GPIO_OUT);
+    gpio_clear(LED1_PIN);
+
+    /* initialize the CPU */
+    cpu_init();
+}
+
+/** @} */
+{% endhighlight %}
+
+{% highlight C %}
+/*
+ * Copyright (C) 2020 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include "cpu.h"
+#include "board.h"
+
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the boards LEDs */
+    gpio_init(LED0_PIN, GPIO_OUT);
+    gpio_clear(LED0_PIN);
+    gpio_init(LED1_PIN, GPIO_OUT);
+    gpio_clear(LED1_PIN);
+
+    /* initialize the CPU */
+    cpu_init();
+}
+
+/** @} */
+{% endhighlight %}
+
+After that I tested if everything worked as configured, [provided some
+documentation how to do that][feather doc], and then called it done.
+
+# Conclusion
+Given that you already worked with C and `make` and that the microcontroller for
+the board you want to port is already provided, porting a board is quite
```

Comment:
you might want to link to the Boards page

Reply:
Why? What context does that provide? Or do you mean the CPU page for "Given […] that the microcontroller for the board you want to port is already provided, […]"

Reply:
CPUs page (and maybe Boards to check whether the Board is already supported).

Reply:
>  (and maybe Boards to check whether the Board is already supported).

Doesn't really fit in this context here. This should be the implied step 1 ;-) (which I won't editorialize in, just to mention the boards page now).

Reply:
Linked the CPUs page.